### DISCUSSION TITLE
targets-specific selection helpers for tar_visnetwork()

### DISCUSSION THREAD

**Discussion by rgayler** [ropensci/targets#474] [Ideas]
The proposal is for selection helpers for `tar_visnetwork()` that are aware of the `targets` DAG structure.

Specifically, `tar_upstream(vertices, max_edge)` would return the union of all vertices no more than `max_edge` edges upstream of each vertex in `vertices`. `tar_downstream(vertices, max_edge)` would do the equivalent for downstream vertices.

These would allow visualisation of the neighbourhood of some vertices of interest.

**Comment by wlandau** [comment]
Just did some digging and found https://stackoverflow.com/questions/39964747/plot-subnetwork-subgraph-after-selecting-nodes-with-visnetwork-in-r. This is pretty cool! Let's take this pipeline:

```r
# _targets.R
library(targets)
library(tarchetypes)
list(
  tar_map(
    values = list(index = seq_len(3)),
    tar_target(a, 1),
    tar_target(b, a),
    tar_target(c, z),
    tar_target(d, c)
  ),
  tar_target(z, c(b_1, b_2, b_3))
)
```

Basic graph:

```r
tar_visnetwork()
```

![Screen Shot 2021-05-29 at 11 03 41 AM](https://user-images.githubusercontent.com/1580860/120075010-8cbb8580-c06d-11eb-91b5-ebbdfb3958a5.png)

Here is what happens when I use `visOptions()` to highlight upstream nodes and then click on `c_2`:

```{r}
library(visNetwork)
tar_visnetwork() %>%
  visOptions(highlightNearest = list(enabled = TRUE, algorithm = "hierarchical", degree = list(from = 2, to = 0)))
```

![Screen Shot 2021-05-29 at 11 05 02 AM](https://user-images.githubusercontent.com/1580860/120075064-c4c2c880-c06d-11eb-9498-91c0b2192dd9.png)

Would be pretty simple to expose arguments to `tar_visnetwork()` and `tar_glimpse()` for the `from` and `to` degrees. I think I will do that.

**Reply by wlandau** [reply]
@rgayler, `tar_visnetwork()` and `tar_glimpse()` now highlight neighborhoods on click as of ec9f98ffecd1d56d91d4e2f6bbf3e68b5606d8ed. See arguments `degree_from` and `degree_to`.

**Reply by rgayler** [reply]
Schmick!
